front/lib/extract_events.ts

```diff
@@ -0,0 +1,122 @@
+import {
+  cloneBaseConfig,
+  DustProdActionRegistry,
+} from "@app/lib/actions/registry";
+import { runAction } from "@app/lib/actions/server";
+import { getInternalBuilderOwner } from "@app/lib/auth";
+import {
+  getRawExtractEventMarkersFromText,
+  sanitizeRawExtractEventMarkers,
+} from "@app/lib/extract_event_markers";
+import { EventSchema, ExtractedEvent } from "@app/lib/models";
+import logger from "@app/logger/logger";
+import { WorkspaceType } from "@app/types/user";
+
+/**
+ * Gets the markers from the doc and calls _processExtractEvent for each of them
+ */
+export async function processExtractEvents(
+  workspaceId: string,
+  documentId: string,
+  documentText: string
+) {
+  const owner = await getInternalBuilderOwner(workspaceId);
+  const rawMarkers = getRawExtractEventMarkersFromText(documentText);
+  const markers = sanitizeRawExtractEventMarkers(rawMarkers);
+
+  await Promise.all(
```

Comment:
How many markers to you expect there can be ? do you expect it to always be small enough to runn all of them in // without issues ?

Reply:
I'm guessing 1 most of the time, but yes nothing prevents someone to add 1000 markers on a doc 😅